Pia 1645 error log

zladzeyka opened the pull request:
### Description
Added error logger (currently print ErrorLog struct)

### How to test

### Merging
Automatic

### Issues
https://ginis.atlassian.net/browse/PIA-1645

zladzeyka committed:
GiniCapture: added Error log structure with default values for system and sdk properties;
GiniCapture: added protocol GiniCaptureErrorLoggerDelegate for default error logging;
GiniConfiguration: added giniErrorLoggerIsOn which is true when default error logging to gini server is on;
Example: updated pods. (PIA-1645)

zladzeyka committed:
Gini Capture: updated capture version (PIA-1645)

zladzeyka committed:
ErrorLog: updated version device model and apiLib version (PIA-1645)

zladzeyka committed:
DocumentService: added error logging (PIA-1645)

zladzeyka committed:
GiniCapture, GiniCapture + Networking: added errorLoggerDelegate in view controller init methods (PIA-1645)

zladzeyka committed:
GiniCapture: added set custom errorLoggerDelegate (PIA-1645)

zladzeyka committed:
GiniConfiguration: added customGiniErrorLoggerDelegate : GiniCaptureErrorLoggerDelegate? property for custom error logging, will work when giniErrorLoggerIsOn = true and GiniCaptureErrorLoggerDelegate? will be implemented (PIA-1645)

zladzeyka committed:
GiniScreenAPICoordinator:
 *  implemented GiniCaptureErrorLoggerDelegate;
 *  implemented Error Logging for CameraViewController. (PIA-1645)

zladzeyka committed:
CameraViewController: added error logger while image capturing (PIA-1645)

zladzeyka committed:
GiniScreenAPICoordinator:
* implemented Error Logging for AnalysisViewController. (PIA-1645)

zladzeyka committed:
GiniScreenAPICoordinator: * fixed error logger delegate set for AnalysisViewController. (PIA-1645)

zladzeyka committed:
GiniNetworkingScreenAPICoordinator: added optional  errorLoggerDelegate in init methods (PIA-1645)

zladzeyka committed:
DocumentService: removed direct errors logger from DocumentService: DocumentServiceProtocol (PIA-1645)

a-szotyori committed:
Centralize error logging logic

Introduced `GiniCaptureErrorLogger` to contain the error logging
logic and avoid having it in multiple places.

a-szotyori commented on GiniCapture/Classes/Core/Logging/ErrorLog.swift:
```diff
@@ -0,0 +1,37 @@
+
+import Foundation
+import GiniPayApiLib
+public struct ErrorLog {
```
Isn't `ErrorLog: Codable` enough without the need for `enum CodingKeys` and the `Decodable` extenstion? The property names are the same.

zladzeyka replied:
ithink it's needed when you don't want to decode /encode everything and for nested types

a-szotyori committed:
Always use error logger instance from `GiniConfiguration`

This makes sure our error logger is called also when using the
component api.

Also using `String(describe: error)` instead of
`error.localizedDescription` because only for `NSErrors` and swift
errors conforming to `LocalizedError` does it return a meaningful
message.

Fixed an async issue in `GiniScreenAPICoordinate+Camera` which
caused the `error.localizedDescription` to be always logged.

a-szotyori reviewed (commented):
Looks good and I have some suggestions.

We should put the logic of when to log to gini and when to use the custom logger into a dedicated class.

Also having two places where to set a custom error logger delegate is confusing. We should only have it in `GiniConfiguration`and remove it from the `viewController()` methods in `GiniCapture`.

Also to make sure our logging delegate is called even when using the component api we should always call our logger from the `GiniConfiguration`.

I created a PR with my suggestions: https://github.com/gini/gini-capture-sdk-ios/pull/23

a-szotyori committed:
Merge pull request #23 from gini/PIA-1645-centralized-logic

Centralize error logging logic

zladzeyka committed:
DocumentService:
 * added error logger for the events: composite doc creation, document creation, deleting document and sending feedback (PIA-1645)
